src/client/dfs/dfs.c

```diff
@@ -850,18 +850,25 @@ open_symlink(dfs_t *dfs, daos_handle_t th, dfs_obj_t *parent, int flags,
 		if (exists)
 			return EEXIST;
 
-		entry.value = (char *)value;
+		if (value == NULL || strnlen(value, PATH_MAX-1) > PATH_MAX-1)
+			return -DER_INVAL;
+
 		rc = oid_gen(dfs, 0, false, &sym->oid);
 		if (rc != 0)
 			return rc;
+		oid_cp(&entry.oid, sym->oid);
 		entry.mode = sym->mode;
 		entry.atime = entry.mtime = entry.ctime = time(NULL);
 		entry.chunk_size = 0;
+		sym->value = strdup(value);
```

Comment:
strndup()

Reply:
since we already checked for size, strdup() should be fine, no?

Reply:
In theory yes, but we still need to avoid using strdup().  You could potentially combine the two calls into one, and call strndup and use the return value to check the length

Reply:
i don't see how that is possible.. strndup doesn't return the length, so if we don't check the length after, we truncate the user value name. An error in this case is better.
I agree we should be more careful with strdup and len, but in this case, it's fine since we validate the size before.

Reply:
You're right about strndup, that's annoying.